packages/create-plugin/src/migrations/utils.ts

```diff
@@ -132,3 +133,174 @@ export function installNPMDependencies(context: Context) {
     execSync(installCmd, { cwd: context.basePath, stdio: 'inherit' });
   }
 }
+
+export function readJsonFile<T extends object = any>(context: Context, path: string): T {
+  if (!context.doesFileExist(path)) {
+    throw new Error(`Cannot find ${path}`);
+  }
+  try {
+    return JSON.parse(context.getFile(path) || '{}');
+  } catch (e) {
+    throw new Error(`Cannot parse ${path}: ${e}`);
+  }
+}
+
+export function addDependenciesToPackageJson(
+  context: Context,
+  dependencies: Record<string, string>,
+  devDependencies: Record<string, string> = {},
+  packageJsonPath = 'package.json'
+) {
+  const currentPackageJson = readJsonFile(context, packageJsonPath);
+  const currentDeps = { ...(currentPackageJson.dependencies || {}) };
+  const currentDevDeps = { ...(currentPackageJson.devDependencies || {}) };
+
+  // Handle dependencies
+  for (const [dep, newVersion] of Object.entries(dependencies)) {
+    if (currentDeps[dep]) {
+      if (isIncomingVersionGreater(newVersion, currentDeps[dep])) {
+        currentDeps[dep] = newVersion;
+      }
+    } else if (currentDevDeps[dep]) {
+      // Exists in devDependencies, only update there if new version is greater
+      if (isIncomingVersionGreater(newVersion, currentDevDeps[dep])) {
+        currentDevDeps[dep] = newVersion;
+      }
+    } else {
+      // Not present, add to dependencies
+      currentDeps[dep] = newVersion;
+    }
+  }
+
+  // Handle devDependencies
+  for (const [dep, newVersion] of Object.entries(devDependencies)) {
+    if (currentDeps[dep]) {
+      // Exists in dependencies, only update there if new version is greater
+      if (isIncomingVersionGreater(newVersion, currentDeps[dep])) {
+        currentDeps[dep] = newVersion;
+      }
+    } else if (currentDevDeps[dep]) {
+      if (isIncomingVersionGreater(newVersion, currentDevDeps[dep])) {
+        currentDevDeps[dep] = newVersion;
+      }
+    } else {
+      // Not present, add to devDependencies
+      currentDevDeps[dep] = newVersion;
+    }
+  }
+
+  // Sort dependencies alphabetically for consistency
+  const sortedDeps = sortObjectByKeys(currentDeps);
+  const sortedDevDeps = sortObjectByKeys(currentDevDeps);
+
+  // Only update if there are actual changes
+  const hasChanges =
+    JSON.stringify(sortedDeps) !== JSON.stringify(currentPackageJson.dependencies || {}) ||
+    JSON.stringify(sortedDevDeps) !== JSON.stringify(currentPackageJson.devDependencies || {});
```

Comment:
Just to make sure I understand correctly - this also updates the package.json if there were no changes in the dependencies but the sorting order has been changed - is that something that we want?